src/dmd/tokens.d

```diff
@@ -499,12 +503,17 @@ static immutable TOK[TOK.max + 1] Ckeywords =
         enum Ckwds = [ auto_, break_, case_, char_, const_, continue_, default_, do_, float64, else_,
                        enum_, extern_, float32, for_, goto_, if_, inline, int32, int64, register,
                        restrict, return_, int16, signed, sizeof_, static_, struct_, switch_, typedef_,
-                       unsigned, void_, volatile, while_,
+                       unsigned, void_, volatile, while_, asm_,
                        _Alignas, _Alignof, _Atomic, _Bool, _Complex, _Generic, _Imaginary, _Noreturn,
                        _Static_assert, _Thread_local, __cdecl, __restrict, __declspec, __attribute__ ];
 
         foreach (kw; Ckwds)
             tab[kw] = cast(TOK) kw;
+
+        // Populate alternate keywords
+        tab[__asm] = asm_;
+        tab[__asm__] = asm_;
```

Comment:
Here we have `asm`, `__asm`, and `__asm__` meaning exactly the same decision. This happens in C compilers because the C spec evolves and fashion evolves, and the old names remain supported.

But do we really want to do this? There is a whole forest of these obsolete keywords. Should we instead just support the most modern one? Let the user use a `#define __asm __asm__` if they want to use the old ones. Thus, we can keep clutter out of ImportC.

Reply:
#12552